### PR TITLE
Improve json export file size

### PR DESCRIPTION
**Pull request by theimo1221**
Hello team,

personally I'm working a lot with the `export all objects as JSON` option, as I use the resulting JSON for other programs/logics as an input file. Over the course of the last few months the exported file kept growing and growing, so I looked into the cause and identified 3 big parts which can greatly reduce the output file size:
1. All News of all Adapters are included, due to `system.repositories` being included.
2. Plenty of states/fields contain translations in all languages
3. JSON Beautify is on, which while improving readability for humans, only slows down machines.

On my machine with an ioBroker Dev-Server only having an ioBroker.ring Instance running the file size could be reduced from 6,2 Mb to 159 Kb.

image

- [ ] Add translations
I'm not quite sure were exactly I have to implement the translations, but if you guys point me in the right direction I'll of course add them.

Best regards

Thiemo

### CODE REVIEW

File: src/src/components/ObjectBrowser.jsx [hunk 1]
```diff
@@ -802,9 +803,36 @@ const styles = theme => ({
     },
 });
 
-function generateFile(filename, obj) {
+function filterObject(obj, filterKeys, excludeTranslations) {
```
Comment (foxriver76) [inline src/src/components/ObjectBrowser.jsx:806]: lets add some jsdoc for the new functions

File: src/src/components/ObjectBrowser.jsx [hunk 2]
```diff
@@ -3578,39 +3609,41 @@ class ObjectBrowser extends Component {
...
+        }
+        if (!(this.state.selected.length || this.state.selectedNonObject)) {
+            window.alert(this.props.t('ra_Save of objects-tree is not possible'));
+            return;
+        }
+        const result = {};
+        const id = this.state.selected[0] || this.state.selectedNonObject;
+        const ids = this._getSelectedIdsForExport();
+
+        for (let i = 0; i < ids.length; i++) {
```
Comment (foxriver76) [inline src/src/components/ObjectBrowser.jsx:3625]:
```suggestion
        for (const key of ids) {
```

saves us one line and is easier to read

File: src/src/components/ObjectBrowser.jsx [hunk 3]
```diff
@@ -3632,13 +3665,40 @@ class ObjectBrowser extends Component {
...
+                            onChange={e => this.setState({ excludeSystemRepositoriesFromExport: e.target.checked })}
+                        />}
+                        label={this.props.t('ra_Exclude system repositories from export JSON')}
+                    />
+                    <FormControlLabel
+                        control={<Checkbox
+                            checked={this.state.excludeTranslations}
+                            onChange={e => this.setState({ excludeTranslations: e.target.checked })}
+                        />}
+                        label={this.props.t('ra_Exclude translations (except english) from export JSON')}
```
Comment (foxriver76) [inline src/src/components/ObjectBrowser.jsx:3690]: you can find the translations here: https://github.com/ioBroker/ioBroker.admin/tree/master/src/src/i18n

Please do not use the `ra_` prefix for the translations managed in this repository.

File: src/src/components/ObjectBrowser.jsx [hunk 4]
```diff
@@ -3578,39 +3609,41 @@ class ObjectBrowser extends Component {
         return [];
     }
 
-    async _exportObjects(isAll, noStatesByExportImport) {
+    async _exportObjects(isAll, noStatesByExportImport, beautifyJsonExport, excludeSystemRepositoriesFromExport, excludeTranslations) {
```
Comment (foxriver76) [inline src/src/components/ObjectBrowser.jsx:3612]: Optional: with so many options (and also many optional) we could think about an options object instead

**Review by foxriver76 — CHANGES_REQUESTED** [review]
lgtm in general, just some small comments

File: src/src/components/ObjectBrowser.jsx (hunk 3 above)
Comment (theimo1221) [reply src/src/components/ObjectBrowser.jsx]: I saw that folder, but as some tools/scripts are used in adapters to populate translations I wasn't sure whether to add it there manually.

@foxriver76 I added translations for german and english, is there a tool/automatism which can fill the other languages at the correct position as well, or do I have to manually fill those?

Comment (foxriver76) [reply src/src/components/ObjectBrowser.jsx]: We are normally using https://github.com/gilmarsquinelato/i18n-manager but needs Google API key. I can add the rest of the translations after merge, no problem.

**Pull request merged**